src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -834,25 +863,24 @@ InsertShardPlacementRow(uint64 shardId, uint64 placementId,
 	{
 		placementId = master_get_new_placementid(NULL);
 	}
-	values[Anum_pg_dist_shard_placement_shardid - 1] = Int64GetDatum(shardId);
-	values[Anum_pg_dist_shard_placement_shardstate - 1] = CharGetDatum(shardState);
-	values[Anum_pg_dist_shard_placement_shardlength - 1] = Int64GetDatum(shardLength);
-	values[Anum_pg_dist_shard_placement_nodename - 1] = CStringGetTextDatum(nodeName);
-	values[Anum_pg_dist_shard_placement_nodeport - 1] = Int64GetDatum(nodePort);
-	values[Anum_pg_dist_shard_placement_placementid - 1] = Int64GetDatum(placementId);
+	values[Anum_pg_dist_placement_placementid - 1] = Int64GetDatum(placementId);
+	values[Anum_pg_dist_placement_shardid - 1] = Int64GetDatum(shardId);
+	values[Anum_pg_dist_placement_shardstate - 1] = CharGetDatum(shardState);
+	values[Anum_pg_dist_placement_shardlength - 1] = Int64GetDatum(shardLength);
+	values[Anum_pg_dist_placement_groupid - 1] = Int64GetDatum(groupId);
 
 	/* open shard placement relation and insert new tuple */
-	pgDistShardPlacement = heap_open(DistShardPlacementRelationId(), RowExclusiveLock);
+	pgDistPlacement = heap_open(DistPlacementRelationId(), RowExclusiveLock);
 
-	tupleDescriptor = RelationGetDescr(pgDistShardPlacement);
+	tupleDescriptor = RelationGetDescr(pgDistPlacement);
 	heapTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
 
-	CatalogTupleInsert(pgDistShardPlacement, heapTuple);
+	CatalogTupleInsert(pgDistPlacement, heapTuple);
 
 	CitusInvalidateRelcacheByShardId(shardId);
 
 	CommandCounterIncrement();
-	heap_close(pgDistShardPlacement, RowExclusiveLock);
+	heap_close(pgDistPlacement, RowExclusiveLock);
```

Comment:
not this PR's fault, but we should keep the `RowExclusiveLock` (replace with `NoLock`)